wsdl2java toolchain article (#238)
karakun/karakun.github.io: _posts/2024-05-28-apache-cxf-toolexception-wsdl2java.md

---
layout: post
title: 'Apache CXF ToolException "Tools plugin provider jaxb context init failed" when using wsdl2java '
authors: ['markus']
featuredImage: gitlab-ci-pipelines
excerpt: "Build a multiproject spanning Pipeline triggered by Tags in Gitlab CI"
permalink: '/2024/05/28/cxf-jaxb-toolexception-wsdl2java.html'
categories: [CXF, JAXB, wsdl2java, XML, SOAP, Code Generation, Gradle]
header:
  text: Apache CXF ToolException with JAXB / wsdl2java 
  image: post
---

> Imagine, in a project you are asked to consume a SOAP webservice. Yes, SOAP is aged as stone BUT it is also very percisely specified as a interface, which make it still a common choice when it comes to intergrate systems. And only integrated systems deliver great benefit on the one hand as well as (old) legacy system have proven their business case which is why they are worth to integrate. (There is *gold* in legacy.) 

Reviewer: I'd actually fully reformulate it as it's a bit hard to follow, but in case you desire to keep it as is, here are some typo corrections.
```suggestion
Imagine, in a project you are asked to consume a SOAP webservice. Yes, SOAP is aged as stone BUT it is also very precisely specified as an interface, which make it still a common choice when it comes to integrate systems. And only integrated systems deliver great benefit on the one hand as well as (old) legacy system have proven their business case which is why they are worth to integrate. (There is *gold* in legacy.) 
```

> So while integrating that SOAP webservice, you try to generate some Java classes from the XML files provided and suddently, you see:

Reviewer: ```suggestion
So while integrating that SOAP webservice, you try to generate some Java classes from the XML files provided and suddenly, you see:
```

}

repositories {
	mavenCentral()
}

ext {
	jaxbApiVersion = "2.2.7" //be careful with Jaxb version changes, xjc relies on it
	jaxbVersion = "2.3.5"
}

dependencies {
	implementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.14'
	implementation 'org.springframework.boot:spring-boot-starter-web'
	implementation 'org.springframework.boot:spring-boot-starter-web-services'
	providedRuntime 'org.springframework.boot:spring-boot-starter-tomcat'
	testImplementation 'org.springframework.boot:spring-boot-starter-test'

	jaxb "com.sun.xml.bind:jaxb-xjc:$jaxbVersion"
	jaxb "com.sun.xml.bind:jaxb-impl:$jaxbVersion"
	jaxb "javax.xml.bind:jaxb-api:$jaxbApiVersion"
}

tasks.named('test') {
	useJUnitPlatform()
}

wsdl2java {
	cxfVersion.set("4.0.2")
	markGenerated.set(true)
	useJakarta.set(true)
	wsdlDir.set(layout.projectDirectory.dir("src/main/resources/service-schema/"))
	generatedSourceDir.set(layout.projectDirectory.dir("build/generated-sources/wsdl2java/"))
}

task copyGeneratedFilesToSourceFolder(type: Copy) {
	dependsOn 'wsdl2java'
	from "${project.buildDir}/generated-sources/wsdl2java"
	include "**/*.java"
	into "${project.projectDir}/src/main/java/"
}
```

> But as i transferred this into the lager project, I got the error mentioned above:

Reviewer: 🍻 
```suggestion
But as i transferred this into the larger project, I got the error mentioned above:
```

```
org.apache.cxf.tools.common.ToolException: Tools plugin provider jaxb context init failed
```

> To get more information, try running my build with the `--stacktrace`flag, but it does not give a more detailled output or more hints than the errormessage above.

Reviewer: ```suggestion
To get more information, tried running my build with the `--stacktrace` flag, but it did not give a more detailed output or more hints than the error message above.
```

> Going one step further with `--debug` , one finds himself flooded with output - but having a closer look and goind through it pay off. We find 

Reviewer: ```suggestion
Going one step further with `--debug`, you find yourself flooded with an output - but having a closer look and going through it payed off. We find 
```
